csrc/scheduler/tools/static_repeat.cpp

```diff
@@ -154,10 +60,39 @@ std::optional<StaticRepeatInfo> getMaybeStaticRepeatInfo(
   }
 
   StaticRepeatInfo info;
-  info.repeat_output_tv = maybe_repeat_out;
-  info.reshape_output_tv = reshape_out;
-  info.reshape_repeat_id = reshape_out->getRootDomain().at(broadcast_pos);
-  info.repeat_tvs = repeat_tvs;
+
+  info.output_id = reshape_merge->out();
+
+  const auto c2p =
+      PairwiseLogicalDomainMap(reshape_in, reshape_out).mapConsumerToProducer();
+
+  auto producer_merge_outer = c2p.at(reshape_merge->outer());
+  auto producer_merge_inner = c2p.at(reshape_merge->inner());
+  IterDomain* producer_factor_id = nullptr;
+
+  if (producer_merge_outer->isBroadcast() &&
+      producer_merge_outer->hasExpandedExtent() &&
+      !producer_merge_inner->isBroadcast()) {
```

Comment:
> !producer_merge_inner->isBroadcast()

Do we need to check for this? What happens when we merge two broadcast ID both with expanded extent?
An earlier comment says

>  54   // Reshape of an expanded broadcast always generates a concrete
 55   // non-broadcast ID, so this check is not necessary, but just in
 56   // case in the future that may change.

Reply:
I think then we should factor out both IDs, so we would want to have a slightly different scheduling approach. I'm not sure if that's worth the additional work.